csharp/src/Apache.Arrow.Flight/StreamWriter.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Grpc.Core;
+
+namespace Apache.Arrow.Flight
+{
+    public class StreamWriter<TIn, TOut> : IAsyncStreamWriter<TIn>
```

Comment:
Does this need to be public? If so, we need to give it a different name, since `StreamWriter` is a common .NET type that we wouldn't want exposed from the Arrow library.

Reply:
changed to internal